src/python/gem5/simulate/simulator.py

```diff
@@ -689,21 +689,14 @@ def run(self, max_ticks: Optional[int] = None) -> None:
                           run. If this ``max_ticks`` value is met, a ``MAX_TICK``
                           exit event is received, if another simulation exit
                           event is met the tick count is reset. This is the
-                          **maximum number of ticks per simulation run**.
-                          **WARNING**: This this parameter is deprecated and
-                          will be removed in future releases of gem5. Please
-                          set the max ticks via the `set_max_ticks` function.
-                          or via the Simulator constructor.
+                          **maximum number of ticks per simulation run.
         """
 
-        if max_ticks:
+        if hasattr(self, "_max_ticks") and self._max_ticks and max_ticks:
```

Comment:
I'm not sure this is going to work as expected. You set `_max_ticks` in the constructor, so `hasattr(self, "_max_ticks")` will always be true. https://github.com/gem5/gem5/blob/9674ca1f45d171e01349d0ae871a52a329fd0667/src/python/gem5/simulate/simulator.py#L293

```suggestion
        if max_ticks and max_ticks != self._max_ticks:
```